### ISSUE TITLE
[rllib] How to Create Loss Function Dependent on global_timestep

### ISSUE THREAD

**Issue by nflu** [ray-project/ray#3280]
I'd like to create a loss function that has a variable that is changed over the course of learning according to the `global_timestep` of the agent (such as [the team-spirit](https://blog.openai.com/openai-five/) hyper-parameter). 

I think to do this I need to be able to access `global_timestep` in the `Optimizer` class. Surely there is some instance attribute in the `Optimizer` class that corresponds to training timesteps but I can't seem to find it. Unless I'm perhaps misunderstanding what `global_timestep` represents.

**Comment by ericl** [comment]
@nflu I think you modify a policy graph instead, since that is where the loss is defined. You can replace some part of the loss with a variable and load new values when on_global_var_update is called. Here's an example of how that works for the built in lr schedules:

https://github.com/ray-project/ray/blob/43df405d07516ba4be8d1a7686c090fde2184b0b/python/ray/rllib/evaluation/tf_policy_graph.py#L278

**Comment by nflu** [comment]
Ah thanks that worked great. Also is there some documentation explaining the relationship between all the various time steps attributes in the different classes?

**Comment by ericl** [comment]
For most algos, they should be the same. It's only with DQN / DDPG where
you have replay and that causes there to be a difference between sampled vs
trained on.
